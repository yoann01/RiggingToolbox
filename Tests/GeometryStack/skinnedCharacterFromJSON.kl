
require RiggingToolbox;

operator entry(){

  StartFabricProfiling();

  String json = "    {\n\
    \"geomOperators\": [
        { \"type\": \"AlembicSkinnedMeshGeometryGenerator\", \"filePath\": \"${FABRIC_RIGGINGTOOLBOX_PATH}/Tests/GeometryStack/Resources/skinnedTube.abc\", \"geometryNames\": [\"pCylinderShape1\"] },
        { \"type\": \"SkinningModifier\" }
    ]
  }";

  GeometryStack stack();
  PersistenceContext persistenceContext();
  stack.loadJSONString(persistenceContext, json);

  SkinningModifier skiningModifier = stack.getGeometryOperator(1);

  EvalContext context();
  stack.evaluate(context);

  // Modify the pose and then reevaluate. 
  Mat44 pose[];
  pose.resize(4);
  pose[0] = Xfo(Vec3(3, 4, 5)).toMat44();
  pose[1] = Xfo(Vec3(10, 20, 5)).toMat44();
  pose[2] = Xfo(Vec3(10, 20, 5)).toMat44();
  pose[3] = Xfo(Vec3(10, 20, 5)).toMat44();
  skiningModifier.setPose(pose);

  GeometrySet geomSet = stack.evaluate(context);

  StopFabricProfiling();

  report( GetEvalPathReport() );

  report("stack:" + stack.getDesc());
  
}

