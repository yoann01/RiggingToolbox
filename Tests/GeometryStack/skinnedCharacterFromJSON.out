Importing:D:/Projects/FabricEngineInc/RiggingToolbox/Tests/GeometryStack/Resources/skinnedTube.abc
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function GeometryCache.update!(io GeometrySet, GeometryOperator)
--function AlembicSkinnedMeshGeometryGenerator.evaluate!(EvalContext, io GeometrySet)
--function GeometryAttributeCache.update!(io GeometrySet, GeometryOperator):["positions"]
----Update:positions
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)
function GeometryStack.notify!(Notifier, String, String):SkinningModifier.changed
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function GeometryAttributeCache.update!(io GeometrySet, GeometryOperator):["positions"]
----Restore:positions
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)

stack:GeometryStack {
  geomOperators:[
    {
      type: AlembicSkinnedMeshGeometryGenerator,
      filePath: ${FABRIC_RIGGINGTOOLBOX_PATH}/Tests/GeometryStack/Resources/skinnedTube.abc
      expandedPath: D:/Projects/FabricEngineInc/RiggingToolbox/Tests/GeometryStack/Resources/skinnedTube.abc
    }
    {
      type: SkinningModifier
    }
  ],
  geomSet:   {
    geometries:[
      pCylinderShape1:{
      },
    ]
    attributeGenerations:{
      positions:2,
    },
  }
}
